modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -985,6 +985,95 @@ public void run() {
         StatusThreadPoolExecutor.getInstance().execute(runnableTask);
     }
 
+    @ReactMethod
```

Comment:
could you please elaborate on why we need these methods in react ?

Reply:
@flexsurfer : we don't actually need the first 3 methods yet only the last one : `deserializeAndCompressKey` 
These got left in because I was experimenting with them to see which one of these methods were able to de compress the community id we receive from desktop to make it into a format we see on mobile..

is okay to keep the unused methods for future or I could delete them now?

Reply:
as far as I know Erik needs one of these methods in his PR

Reply:
yes @siddarthkay I'll need to convert "z..." compresssed keys to uncompressed "0x..." keys. thanks!

Reply:
could you please elaborate on why do we need to do this in react ?

Reply:
Sure, @flexsurfer 
lets say you make a new community in desktop 
and you click on share and you get a link like this : https://join.status.im/c/zQ3shTAten2v9CwyQD1Kc7VXAqNPDcHZAMsfbLHCZEx6nFqk9
Then when you paste this link in chat on mobile to another user,
Our existing react code picks up zQ3shTAten2v9CwyQD1Kc7VXAqNPDcHZAMsfbLHCZEx6nFqk9  as the community id and searches for it in the mailserver ( if it is not already a part of our db ).

This community is not found because zQ3shTAten2v9CwyQD1Kc7VXAqNPDcHZAMsfbLHCZEx6nFqk9  is a short key ( some people call it compressed key or serialized key )

hence we need to convert it to a community id that can be searchable which begins with 0x02 format.
To achieve this we need to pass this short key to status go and get back the format which allows us to discover and join communities.

Reply:
that sounds strange,`that can be searchable which begins with 0x02 format.` why can't it be searchable with the key from desktop? where do we search ?

Reply:
I do not know why things are the way they are, but the search code is here: https://github.com/status-im/status-go/pull/2879#discussion_r1058222003

Reply:
it would be ideal if all the keys followed same formats and were consistent, but my hypothesis is that status-desktop team updates the format of the key on the client side to make it appear shorter, else they should have been searchable by the mailserver.

If that was not the case then we would have never received this issue : https://github.com/status-im/status-mobile/issues/14560

Reply:
@flexsurfer : Desktop does this on client side : https://github.com/status-im/status-desktop/pull/5269

Reply:
omg why do we need this headache, short keys have zero sense :( i already had lots of issues restoring my accounts from a mobile on desktop, I know its a rhetorical question, *just crying in the corner*

Reply:
Well this PR just makes sure that we are able to understand short keys, i.e join a community which was made by desktop app. 
This PR does not introduce short key format for communities in mobile UI. 

Also this capability will be essential when we do RC1 release and mobile users try to join the dog-fooding communities made by Desktop :)